packages/react/src/components/DateTimePicker/dateTimePickerUtils.js

```diff
@@ -96,3 +101,483 @@ export const parseValue = (timeRange, dateTimeMask, toLabel) => {
+      return Array.from(siblings).filter(
+        (sibling) => parseInt(sibling.getAttribute('tabindex'), 10) !== -1
+      );
+    }
+
+    return [];
+  };
+
+  const onFieldInteraction = ({ key }) => {
+    switch (key) {
+      case 'Escape':
+        setIsExpanded(false);
+        break;
+      // if the input box is focused and a down arrow is pressed this
+      // moves focus to the first item in the preset list that has a tabindex
+      case 'ArrowDown':
+        /* istanbul ignore else */
+        if (presetListRef?.current) {
+          const listItems = getFocusableSiblings();
+          /* istanbul ignore else */
+          if (listItems?.[0]?.focus) {
+            listItems[0].focus();
+          }
+        }
+        break;
+      default:
+        setIsExpanded(!isExpanded);
+        break;
+    }
+  };
+
+  /**
+   * Moves up the preset list to the previous focusable element or wraps around to the bottom
+   * if already at the top.
+   */
+  const moveToPreviousElement = () => {
+    const siblings = getFocusableSiblings();
+    const index = siblings.findIndex((elem) => elem === document.activeElement);
+    const previous = siblings[index - 1];
+    if (previous) {
+      previous.focus();
+    } else {
+      siblings[siblings.length - 1].focus();
+    }
+  };
+
+  /**
+   * Moves down the preset list to the next focusable element or wraps around to the top
+   * if already at the bottom
+   */
+  const moveToNextElement = () => {
+    const siblings = getFocusableSiblings();
+    const index = siblings.findIndex((elem) => elem === document.activeElement);
+    const next = siblings[index + 1];
+    if (next) {
+      next.focus();
+    } else {
+      siblings[0].focus();
+    }
+  };
+
+  const onNavigatePresets = ({ key }) => {
+    switch (key) {
+      case 'ArrowUp':
+        moveToPreviousElement();
+        break;
+      case 'ArrowDown':
+        moveToNextElement();
+        break;
+      default:
+        break;
+    }
+  };
+
+  /**
+   * Allows navigation back and forth between the radio buttons for Relative/Absolute
+   *
+   * @param {KeyboardEvent} e
+   */
+  const onNavigateRadioButton = (e) => {
+    if (e.target.getAttribute('id').includes('absolute')) {
+      setCustomRangeKind(PICKER_KINDS.RELATIVE);
+      document.activeElement.parentNode.previousSibling
+        .querySelector('input[type="radio"]')
+        .focus();
+    } else {
+      setCustomRangeKind(PICKER_KINDS.ABSOLUTE);
+      document.activeElement.parentNode.nextSibling.querySelector('input[type="radio"]').focus();
+    }
+  };
+
+  return {
+    presetListRef,
+    isExpanded,
+    setIsExpanded,
+    getFocusableSiblings,
+    onFieldInteraction,
+    onNavigateRadioButton,
+    onNavigatePresets,
+  };
+};
+
+/**
+ * Get an alternative human readable value for a preset to show in tooltips and dropdown
+ * ie. 'Last 30 minutes' displays '2020-04-01 11:30 to Now' on the tooltip
+ * @param {Object} object an object containing:
```

Comment:
typo
```suggestion
 * @param {Object} an object containing:
```

Reply:
That's actually standard jsdoc, I just didn't give the parameter a name and defaulted to object since it's destructured. Although, in looking it up I found this syntax for destructured parameters that I'd never seen before. https://jsdoc.app/tags-param.html#parameters-with-properties That's a little nicer than what I did here.